src/multidim/intervalbox.jl

```diff
@@ -3,53 +3,56 @@
 """An `IntervalBox` is an `N`-dimensional rectangular box, given
 by a Cartesian product of `N` `Interval`s.
 """
-struct IntervalBox{N,T} <: StaticVector{N, Interval{T}}
-    data::NTuple{N,Interval{T}}
+struct IntervalBox{N,T}
+    v::SVector{N,Interval{T}}
 end
 
-# StaticArrays.Size{N,T}(::Type{IntervalBox{N,T}}) = StaticArrays.Size(N) # @pure not needed, I think...
-Base.@propagate_inbounds Base.getindex(a::IntervalBox, i::Int) = a.data[i]
+# IntervalBox(x::Interval) = IntervalBox( SVector(x) )  # single interval treated as tuple with one element
 
+IntervalBox(x::Interval...) = IntervalBox(SVector(x))
+IntervalBox(x::Tuple{T}) where {T<:Interval} = IntervalBox(SVector(x))
 
-IntervalBox(x::Interval) = IntervalBox( (x,) )  # single interval treated as tuple with one element
-
+Base.getindex(X::IntervalBox, i) = X.v[i]
```

Comment:
This replaces what it was in line 11, before; do we need `Base.@propagate_inbounds` here, or not?

Reply:
Good question. To be honest I'm not sure, but it probably doesn't hurt.

Reply:
I'm not sure either... I've never understood why it was needed. Proceed as you prefer.